cfme/networks/security_group.py

```diff
@@ -54,7 +54,16 @@ def all(self):
         else:
             view = navigate_to(self, 'All')
         list_networks_obj = view.entities.get_all(surf_pages=True)
-        return [self.instantiate(name=s.name) for s in list_networks_obj]
+
+        collection = (self.instantiate(name=s.name) for s in list_networks_obj)
+        return list(a for a in collection if apply_filters(a, self.filters))
+
+
+def apply_filters(item, filters):
```

Comment:
Required:
Please move it to `collection class` and restrict the users from direct use by changing the method name to  `_apply_filters()`